### PR TITLE
Feat: add screen_hint param in /authorize api for explicit signup redirection

### PR DESCRIPTION
**Pull request by scaletech-milan**
#### What does this PR do?
Enable signup redirection via screen_hint param in /authorize API

### CODE REVIEW

File: server/handlers/authorize.go [hunk 1]
```diff
@@ -120,27 +121,33 @@ func AuthorizeHandler() gin.HandlerFunc {
...
 		} else {
-			loginState += "&nonce=" + nonce
+			authState += "&nonce=" + nonce
 			if err := memorystore.Provider.SetState(state, nonce); err != nil {
 				log.Debug("Error setting temp code", err)
 			}
 		}
 
-		loginURL := "/app?" + loginState
+		authURL := "/app?" + authState
```
Comment (lakhansamani) [inline server/handlers/authorize.go:137]: Good idea is to not put `?` & `#` in authURL use it or declare another variable which just holds path `/app`or `/app/signup` and use that variable in line 140, 144, 146

**Review by lakhansamani — CHANGES_REQUESTED** [review]
This PR is missing change in signup page which should also include social media buttons

**Comment by scaletech-milan** [comment]
> This PR is missing change in signup page which should also include social media buttons

File: server/handlers/authorize.go (hunk 1 above)
Comment (scaletech-milan) [reply server/handlers/authorize.go]: Updated the changes and added social media buttons on signup page

File: server/constants/oauth2.go [hunk 2]
```diff
@@ -16,4 +16,7 @@ const (
 	ResponseTypeToken = "token"
 	// For the Implicit grant of id_token, use response_type=id_token to include an identifier token.
 	ResponseTypeIDToken = "id_token"
+
+	// Constant indicating the "signup" screen hint for customizing authentication process and redirect to a signup page.
+	ScreenHint = "signup"
```
Comment (lakhansamani) [inline server/constants/oauth2.go:21]: Lets rename this const to `ScreenHintSignUp`
If there is something else available for screen hint then also we can accomodate

**Review by lakhansamani — APPROVED** [review]
Just a minor suggestion overall looks good

**Review by lakhansamani — APPROVED** [review]

File: server/constants/oauth2.go (hunk 2 above)
Comment (scaletech-milan) [reply server/constants/oauth2.go]: Yes True,
Updated!

**Pull request merged**